ecosystem/indexer-grpc/indexer-grpc-table-info/src/internal_indexer_db_service.rs

```diff
@@ -15,17 +16,25 @@ pub struct InternalIndexerDBService {
 }
 
 impl InternalIndexerDBService {
-    pub fn new(db_reader: Arc<dyn DbReader>, node_config: &NodeConfig) -> Self {
+    pub fn new(
+        db_reader: Arc<dyn DbReader>,
+        node_config: &NodeConfig,
+        internal_indexer_db: Option<Arc<DB>>,
+    ) -> Self {
         let db_path = node_config
             .storage
             .get_dir_paths()
             .default_root_path()
             .join(INTERNAL_INDEXER_DB);
         let rocksdb_config = node_config.storage.rocksdb_configs.index_db_config;
```

Comment:
nit: seems duplicated code, something like this will do? 

And why can't we always require the db to be passed in?

```
let db = if let Some(db) = internal_indexer_db {
    db
} else {
    Self::get_indexer_db(node_config).expect() 
}
```